arch/riscv/core/isr.S

```diff
@@ -162,6 +180,10 @@ SECTION_FUNC(exception.entry, _isr_wrapper)
 	DO_CALLER_SAVED(sr)		;
 #endif /* CONFIG_RISCV_SOC_HAS_ISR_STACKING */
 
+#ifdef CONFIG_EXTRA_EXCEPTION_INFO
+	DO_CALLEE_SAVED(sr)		;
```

Comment:
this just introduces a useless burden of saving and restoring callee saved registers every single time even when no context switch is actually happening. If the goal here is to print the content of the callee-saved registers on fatal error, you can save the registers just before calling `z_riscv_fatal_error` considering that the assembly code is not touching any of these registers.

Reply:
Thanks for the review, I have moved it to right above the `z_riscv_fatal_error`